DQMOffline/JetMET/src/METAnalyzer.cc

```diff
@@ -1504,14 +1504,17 @@ void METAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetu
   //met=&(tcmetcoll->front());
   //}
   if (isPFMet_) {
+    assert(!pfmetcoll->empty());
```

Comment:
@Dr15Jones @jfernan2 are we expecting this instruction to abort the job? Is this the behaviour that we want in this DQM analyzer? Or should we set a different type of protection?

Reply:
The lines following lead to undefined behavior. In the case of programming errors I use `assert`. In the case of runtime problems I'd use exceptions. At the cursory glance I gave the code, it looked like if you got here with an empty collection it was a logic problem with the algorithm. Hence why I chose `assert`.